tasks/go.py

```diff
@@ -428,3 +435,182 @@ def go_fix(ctx, fix=None):
+    - packages: Comma separated list of packages the will use the new module
+    """
+
+    path = path.rstrip('/')
+
+    if check_uncommitted_changes(ctx):
+        raise RuntimeError("There are uncomitted changes, all changes must be committed to run this command.")
+
+    # Perform checks + save current state to restore it in case of failure
+    assert not os.path.exists(path + '/go.mod'), f"Path {path + '/go.mod'} already exists"
+    is_empty = not os.path.exists(path)
+
+    # Get info
+    with open('go.mod') as f:
+        mainmod = f.read()
+
+    goversion_regex = re.compile(r'^go +([.0-9]+)$', re.MULTILINE)
+    goversion = next(goversion_regex.finditer(mainmod)).group(1)
+
+    # Module content
+    gomod = f"""
+    module github.com/DataDog/datadog-agent/{path}
+
+    go {goversion}
+    """.replace('    ', '')
+
+    try:
+        # Create package
+        print(color_message(f"Creating package {path}", "blue"))
+
+        ctx.run(f"mkdir -p {path}")
+        with open(f"{path}/go.mod", 'w') as f:
+            f.write(gomod)
+
+        if not is_empty:
+            # 1. Update current module
+            deps = get_deps(ctx, path)
+            add_replaces(ctx, path, deps)
+            with ctx.cd(path):
+                ctx.run('go mod tidy')
+
+            # Find and update indirect replaces within go.mod
+            with open(path + '/go.mod') as f:
+                mod_content = f.read()
+                replaces = {
+                    replace
+                    for replace in re.findall(r'github.com/DataDog/datadog-agent/([^\n ]*)', mod_content)
+                    if replace != path
+                }
+            with open(f"{path}/go.mod", 'w') as f:
+                # Cancel mod tidy since it can update the go version
+                f.write(gomod)
+            add_replaces(ctx, path, replaces)
+            with ctx.cd(path):
+                ctx.run('go mod tidy')
+
+            # 2. Update dependencies
+            # Find module that must include the new module
+            dependent_modules = []
+            for gomod in glob.glob('./**/go.mod', recursive=True):
+                mod_path = os.path.dirname(gomod)
+                if os.path.abspath(mod_path) != os.path.abspath(path):
+                    deps = get_deps(ctx, mod_path)
+                    if path in deps:
+                        dependent_modules.append(mod_path)
+
+            for mod in dependent_modules:
+                add_replaces(ctx, mod, [path])
+
+        # Update modules.py
+        add_go_module(path)
+
+        if not is_empty:
+            # Tidy all
+            print(color_message("Running tidy-all task", "bold"))
+            tidy_all(ctx)
+
+        if not no_verify:
+            # Stage updated files since some linting tasks will require it
+            print(color_message("Staging new module files", "bold"))
+            ctx.run("find . '(' -name modules.py -o -name go.mod -o -name go.sum ')' -exec git add '{}' ';'")
```

Comment:
I think this does not work on Windows but I'm not sure. Could you check if we need to support Windows?

Reply:
git add . works